list.go

```diff
@@ -0,0 +1,110 @@
+// +build linux
+
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strconv"
+	"syscall"
+	"text/tabwriter"
+	"time"
+
+	"github.com/Sirupsen/logrus"
+	"github.com/codegangsta/cli"
+	"github.com/opencontainers/runc/libcontainer/system"
+)
+
+var listCommand = cli.Command{
+	Name:  "list",
+	Usage: "lists containers started by runc with the given root",
+	Action: func(context *cli.Context) {
+
+		// preload the container factory
+		if factory == nil {
+			err := factoryPreload(context)
+			if err != nil {
+				logrus.Fatal(err)
+				return
+			}
+		}
+
+		// get the list of containers
+		root := context.GlobalString("root")
+		absRoot, err := filepath.Abs(root)
+		if err != nil {
+			logrus.Fatal(err)
+			return
+		}
+		list, err := ioutil.ReadDir(absRoot)
+
+		w := tabwriter.NewWriter(os.Stdout, 12, 1, 3, ' ', 0)
+		fmt.Fprint(w, "ID\tSTARTED\tSTATUS\tPID\tROOTFS\n")
+
+		// output containers
+		for _, item := range list {
+			switch {
+			case !item.IsDir():
+				// do nothing with misc files in the containers directory
+			case item.IsDir():
+				outputListInfo(item.Name(), w)
+			}
+		}
+
+		if err := w.Flush(); err != nil {
+			logrus.Fatal(err)
+		}
+	},
+}
+
+func outputListInfo(id string, w *tabwriter.Writer) {
+	container, err := factory.Load(id)
+	if err != nil {
+		logrus.Fatal(err)
+		return
+	}
+
+	containerStatus, err := container.Status()
+	if err != nil {
+		logrus.Fatal(err)
+		return
+	}
+
+	state, err := container.State()
+	if err != nil {
+		logrus.Fatal(err)
+		return
+	}
+
+	conf := container.Config()
+
+	// convert the InitProcessStartTime string (which is really clock ticks since
+	// boot) to seconds since boot (have to divide by clock ticks)
+	pidStarted, err := strconv.ParseInt(state.BaseState.InitProcessStartTime, 10, 64)
+	if err != nil {
+		logrus.Fatal(err)
+		return
+	}
+	pidStarted /= int64(system.GetClockTicks())
+
+	//grab sysinfo to identify current uptime
+	sysinfo := syscall.Sysinfo_t{}
+	err = syscall.Sysinfo(&sysinfo)
+	if err != nil {
+		logrus.Fatal(err)
+		return
+	}
+
+	// running for is the difference between pidStarted and current uptime
+	// Note: time.Duration is in nanoseconds
+	runningFor := time.Duration((sysinfo.Uptime - pidStarted) * 1000 * 1000 * 1000)
```

Comment:
I find the use of duration here less helpful than the actual time itself. While clocks will vary between systems, at least with a concrete time you know what exact time something started on that system.  With a duration I now need to know when my query was executed and take that into account to get the real date/time.  Let use a time instead.